src/components/Filters/components/TagsInput/TagsInput.tsx

```diff
@@ -127,56 +131,63 @@ const TagsInput: React.FC<TagsInputProps> = ({
     }
   };
 
-  const handleOnChange = (event) => {
+  const handleOnValidate = (event) => {
     const { target } = event;
-    const { value, validity } = target;
 
-    if (patternMessage) {
-      target.setCustomValidity(validity.patternMismatch ? patternMessage : '');
-    }
-    setIsInvalid(validity.patternMismatch);
+    const hasError = onValidate(target) && isNonEmptyString(target.value);
+    onError(hasError);
+  };
+
+  const handleOnChange = (event) => {
+    const {
+      target: { value },
+    } = event;
     setInput(value);
+    handleOnValidate(event);
   };
 
   const placeholderText = tags.length === 0 ? placeholder : '';
 
   const handleOnBlur = (event) => {
-    const { target } = event;
-    const { value, validity } = target;
+    const {
+      target: { value },
+    } = event;
 
-    if (value && !validity.patternMismatch) {
+    if (value && !isInvalid) {
```

Comment:
I'm able to add invalid tag value on blur :(

Reply:
But Apply button is disabled so we can fix that later, imo

Reply:
Story "Tags Input Validation", insert wrong value and click outside the input. Tag with incorrect value is added. You can compare this behavior with adding of wrong value on enter.

Reply:
I don't have this behavior on blur  :S

Reply:
It is fixed